app/(pages)/_contexts/SelectModeContext.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { createContext, useContext, useState, ReactNode } from 'react';
+
+interface SelectModeContextType {
+  selectMode: boolean;
+  toggleSelectMode: () => void;
+  setSelectMode: (mode: boolean) => void;
+}
+
+const SelectModeContext = createContext<SelectModeContextType | undefined>(
```

Comment:
Rather than undefined, maybe just input some default value that fits the SelectModeContextType interface. If this is not possible, then this is fine.